crates/trie/sparse/benches/update.rs

```diff
@@ -0,0 +1,79 @@
+#![allow(missing_docs)]
+
+use alloy_primitives::{B256, U256};
+use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion};
+use proptest::{prelude::*, strategy::ValueTree};
+use reth_trie_common::Nibbles;
+use reth_trie_sparse::SparseTrie;
+
+const LEAF_COUNTS: [usize; 3] = [100, 1_000, 5_000];
+
+fn update_leaf(c: &mut Criterion) {
+    let mut group = c.benchmark_group("update_leaf");
+
+    for leaf_count in LEAF_COUNTS {
+        group.bench_function(BenchmarkId::from_parameter(leaf_count), |b| {
+            let leaves = generate_leaves(leaf_count);
+            b.iter_with_setup(
+                || {
+                    // Start with an empty trie
+                    let mut trie = SparseTrie::revealed_empty();
+                    // Pre-populate with data
+                    for (path, value) in &leaves {
+                        trie.update_leaf(path.clone(), value.clone()).unwrap();
+                    }
+                    trie
+                },
+                |mut trie| {
+                    // Update existing leaves with new values
+                    for (i, (path, _)) in leaves.iter().enumerate().take(100) {
+                        let new_value = alloy_rlp::encode_fixed_size(&U256::from(i * 2)).to_vec();
+                        trie.update_leaf(path.clone(), new_value).unwrap();
+                    }
+                    trie
```

Comment:
should this encoding be part of the setup? ie construct the enumerate iterator, map to path / new_value, then the loop body would not have any encoding? and getting rid of the clone would be nice, that way we can just have a loop and `update_leaf` in the benchmarked fn